config/configmapprovider/properties.go

```diff
@@ -25,31 +24,26 @@ import (
 	"go.opentelemetry.io/collector/config"
 )
 
-type propertiesMapProvider struct {
-	properties []string
-}
-
-// NewProperties returns a Provider, that provides a config.Map from the given properties.
+// NewPropertiesConverter returns a service.ConfigMapConverterFunc, that overrides all the given properties into the
+// input map.
 //
 // Properties must follow the Java properties format, key-value list separated by equal sign with a "."
 // as key delimiter.
 //  ["processors.batch.timeout=2s", "processors.batch/foo.timeout=3s"]
-// TODO: Change this to be a service.ConfigMapConverterFunc.
-func NewProperties(properties []string) Provider {
-	return &propertiesMapProvider{
-		properties: properties,
+// TODO: Find the right place for service.ConfigMapConverterFunc and return that. Currently this will cause circular dependency.
+func NewPropertiesConverter(properties []string) func(*config.Map) error {
```

Comment:
Why do we want this to be a converter? I'm not saying it cannot be, but it still can be considered as a source of config map... 

Maybe `MergePropertiesConverter` in that case?

Reply:
Override? since it is clear. The "Provider" now accepts a "location" (see https://github.com/open-telemetry/opentelemetry-collector/pull/4657) a.k.a an "url" or something similar from where to retrieve the config, here it is just overriding some values, it does not fetch any source to retrieve a config.

Reply:
`OverridePropertiesConverter` sounds good!

Reply:
I think correctly is "overwrite" since we replace that:

To **"overwrite"** something is to put something else in its place, destroying the thing overwritten. To **"override"** something is to cause something else to operate instead of it without harming or changing the thing overridden.